### PR TITLE
Allow specifying filename, line and log ID.

### PR DESCRIPTION
**Pull request by qwandor-google**
Add declarations for __android_log_is_loggable,
__android_log_write_log_message and related types.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -30,6 +30,35 @@ pub enum LogPriority {
     SILENT = 8,
 }
 
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy)]
+#[repr(i32)]
+pub enum log_id_t {
```
Comment (Dushistov) [inline src/lib.rs:36]: may be mark it as non_exhaustive, if it may be changed in future?

File: src/lib.rs [hunk 2]
```diff
@@ -30,6 +30,35 @@ pub enum LogPriority {
     SILENT = 8,
 }
 
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy)]
+#[repr(i32)]
```
Comment (Dushistov) [inline src/lib.rs:35]: Why i32? I don't see any any type hint about size of this enum in C code,
so it should be <= sizeof(int) and that's all we know about it according to https://stackoverflow.com/questions/366017/what-is-the-size-of-an-enum-in-c

File: src/lib.rs [hunk 3]
```diff
@@ -30,6 +30,35 @@ pub enum LogPriority {
...
+    SECURITY = 6,
+    KERNEL = 7,
+    MAX = 8,
+    DEFAULT = 0x7FFFFFFF,
+}
+
+#[allow(non_camel_case_types)]
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct __android_log_message {
```
Comment (Dushistov) [inline src/lib.rs:52]: I found documentation, but can not find real definition of this struct in log.h in my android NDK.
Is it new feature, or deprecated feature?

File: src/lib.rs (hunk 1 above)
Comment (qwandor-google) [reply src/lib.rs:37]: Done.

File: src/lib.rs (hunk 2 above)
Comment (qwandor-google) [reply src/lib.rs:36]: It's used (only) for the `__android_log_message.buffer_id` field, which is declared as int32_t. This is the same type as is used for LogPriority.

File: src/lib.rs (hunk 3 above)
Comment (qwandor-google) [reply src/lib.rs:53]: It was added in API level 30 (i.e. Android 11): https://android-review.googlesource.com/c/platform/system/core/+/1253126

**Comment by Dushistov** [comment]
@qwandor-google 

I am ok with this patch.
Let's wait a couple of days for the reaction of @Nercury .
If there are no objections or reaction from him I will merge PR.

**Comment by Nercury** [comment]
All good, thanks!

**Pull request merged**